src/FffGcodeWriter.cpp

```diff
@@ -107,6 +111,75 @@ void FffGcodeWriter::writeGCode(SliceDataStorage& storage, TimeKeeper& time_keep
 }
 
+void FffGcodeWriter::findLayerSeamsForSpiralize(SliceMeshStorage& mesh)
+{
+    // The spiral has to continue on in an anti-clockwise direction from where the last layer finished, it can't jump backwards
+
+    if (mesh.layers.size() > 0)
+    {
+        if(mesh.layers[0].parts.size() > 0 && mesh.layers[0].parts[0].insets.size() > 0)
+        {
+            // use the vertex closest to Point(0, 0) for the first layer
+            // Doing this, we give the user some degree of control as to where the seam will be.
+            // By moving the model on the bed they can move the start position of the seam and
+            // this could be useful if the spiralization has a problem with a particular seam position
+            mesh.layers[0].seam_vertex_index = PolygonUtils::findClosest(Point(0, 0), mesh.layers[0].parts[0].insets[0][0]).point_idx;
+        }
+
+        for (unsigned layer_nr = 1; layer_nr < mesh.layers.size(); ++layer_nr)
+        {
+            SliceLayer& layer = mesh.layers[layer_nr];
+            const SliceLayer& last_layer = mesh.layers[layer_nr - 1];
+
+            if (layer.parts.size() < 1 || layer.parts[0].insets.size() < 1 || last_layer.parts.size() < 1 || last_layer.parts[0].insets.size() < 1)
+            {
+                // either this layer or the last has no parts (or a part has no insets) so just duplicate the index calculated for the last layer
+                layer.seam_vertex_index = last_layer.seam_vertex_index;
+                continue;
+            }
+            PolygonRef last_wall = last_layer.parts[0].insets[0][0];
+            PolygonRef wall = layer.parts[0].insets[0][0];
+            const int n_points = wall.size();
+            Point last_wall_seam_vertex = last_wall[last_layer.seam_vertex_index];
+
+            // seam_vertex_idx is going to be the index of the seam vertex in the current wall polygon
+            // initially we choose the vertex that is closest to the seam vertex in the last spiralized layer processed
+
+            int seam_vertex_idx = PolygonUtils::findClosest(last_wall_seam_vertex, wall).point_idx;
+
+            // now we check that the vertex following the seam vertex is not to the right of the seam vertex in the last layer
+            // and if it is we move forward
+
+            // get the inward normal of the last layer seam vertex
+            Point last_wall_seam_vertex_inward_normal = PolygonUtils::getVertexInwardNormal(last_wall, last_layer.seam_vertex_index);
+
+            // create a vector from the normal so that we can then test the vertex following the candidate seam vertex to make sure it is on the correct side
+            Point last_wall_seam_vertex_vector = last_wall_seam_vertex + last_wall_seam_vertex_inward_normal;
+
+            // now test the vertex following the candidate seam vertex and if it lies to the left or on the vector, it's good to use
```

Comment:
Note that the first point to the left of the vector might be pretty far away if the geometry of the next layer is rather different from the last.
Perhaps you should also check for the distance between the last wall seam and the seam vertex, compared to the distance between the last wall seam and the first seam vertex.
If that distance is relatively large, you should revert to using the closest point.

I'm not totally sure you should do this, though. This might be optimizing an edge case for which spiralize doesn't work so good in any case...